specs/altair/beacon-chain.md

```diff
@@ -71,7 +71,7 @@ Altair is the first beacon chain hard fork. Its main features are:
 
 | Name | SSZ equivalent | Description |
 | - | - | - |
-| `ParticipationFlags` | `uint8` | a succinct representation of 8 boolean participation flags |
+| `ParticipationFlags` | `byte` | a succinct representation of 8 boolean participation flags |
```

Comment:
Changing this changes the JSON representation, which would be a significant pain for libraries that parse the current output.  This may be considered a wart, but I'm not a fan of changing specs retrospectively that can have an impact on what is right now a working system.

Reply:
> Changing this changes the JSON representation

This is a beacon API concern mostly, ie from the point of view of this spec, nothing changes (ie all SSZ clients remain compatible).

In terms of beacon API, it has a number of possibilities due to the fact that it's not used in any regular API (only debug):

* Leave the wart and cite it as an exception to the canonical JSON spec
* When moving the API out of the `debug` namespace with JSON support, fix the wart
* When moving the API out of the `debug` namespace, drop support for JSON entirely.

Reply:
Or we could leave the spec without the change.  What is the benefit here to changing the data type at this stage?

Reply:
assuming people agree flags are not and were not meant to be integers, it would clarify what the flag is now that we have the language to express it. 

Notably, we're not changing the flags field - we're disambiguating what previously was considered an alias (ie no difference) to accommodate the increased level of information.

The SSZ spec never made any distinction - the beacon api spec independently did.

Reply:
removed this change in the interest of unblocking this PR - it can be done separately if desired